### PR TITLE
Levels

### PR DESCRIPTION
**Pull request by danalex97**
This is custom level generation. Most of it is only additions. Some stuff is moved around from map_generator as otherwise we would have had circular imports.

### CODE REVIEW

File: aimmo-game/simulation/avatar/avatar_manager.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 from simulation.avatar.avatar_appearance import AvatarAppearance
 import copy
 
+import copy
```
Comment (mikebryant) [inline aimmo-game/simulation/avatar/avatar_manager.py]: Remove

File: aimmo-game/simulation/custom_map.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import abc
+
+from simulation.levels.levels import LEVELS
+from pprint import pprint
```
Comment (mikebryant) [inline aimmo-game/simulation/custom_map.py]: Unused, remove?

File: aimmo-game/simulation/levels/parsers.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import json
+import abc
+import os
+
+from pprint import pprint
```
Comment (mikebryant) [inline aimmo-game/simulation/levels/parsers.py]: Unused, remove?

**Comment by danalex97** [comment]
Useful link: https://github.com/ocadotechnology/aimmo/wiki/Levels

File: aimmo-game/simulation/levels/models/objects.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+[
+  {
+    "code": "2",
+    "id" : "class:CellTransform.compute_id",
+    "x" : "class:CellTransform.get_x",
+    "y" : "class:CellTransform.get_y",
+    "value" : {
+      "dsadsa" : "2"
```
Comment (CelineBoudier) [inline aimmo-game/simulation/levels/models/objects.json]: yes?

Comment (danalex97) [reply aimmo-game/simulation/levels/models/objects.json]: Yeah, that is useless :+1:  good catch

File: aimmo-game/simulation/avatar/avatar_manager.py [hunk 5]
```diff
@@ -27,9 +27,7 @@ def get_avatar(self, user_id):
         return self.avatars_by_id[user_id]
 
     def remove_avatar(self, user_id):
-        if user_id in self.avatars_by_id:
-            self.avatars_to_delete_by_id[user_id] = copy.deepcopy(self.avatars_by_id[user_id])
-            del self.avatars_by_id[user_id]
+        del self.avatars_by_id[user_id]
```
[the remark this answers is not in the document]
Comment (danalex97) [reply aimmo-game/simulation/avatar/avatar_manager.py]: You are right, it should be checked.  It should still be checked.

File: aimmo-game/simulation/avatar/avatar_manager.py [hunk 6]
```diff
@@ -28,7 +28,6 @@ def get_avatar(self, user_id):
 
     def remove_avatar(self, user_id):
         if user_id in self.avatars_by_id:
-            self.avatars_to_delete_by_id[user_id] = copy.deepcopy(self.avatars_by_id[user_id])
```
Comment (CelineBoudier) [inline aimmo-game/simulation/avatar/avatar_manager.py]: is this self.avatars_to_delete_by_id still used/useful in the avatar manager?

Comment (danalex97) [reply aimmo-game/simulation/avatar/avatar_manager.py]: This is also done in #223 I will remove it here.

Comment (danalex97) [reply aimmo-game/simulation/avatar/avatar_manager.py]: It should have been called, actually. I got confused as we don't do it anymore in #223

**Pull request merged**